assets/app/views/editor/edit-file.js

```diff
@@ -246,14 +299,94 @@ var EditorView = Backbone.View.extend({
     }
 
     return this;
-  }
-});
+  },
+  saveNewDocument: function () {
+    var self = this;
+
+    try {
+      pageTitle = this.fileNameFromTitle(yaml.parse(this.settings).title);
+    } catch (error) {
+      pageTitle = (new Date()).getTime().toString();
+    }
+
+    pageTitle = [pageTitle.replace(/\W/g, '-'), 'md'].join('.');
+
+    this.listenToOnce(this.model, 'github:commit:success', function(m){
+      var owner = self.model.get('owner'),
+          repoName  = self.model.get('repoName'),
+          branch = self.model.get('branch'),
+          url = ['#edit', owner, repoName, branch, m.request.path].join('/');
 
-function fileNameFromTitle (title) {
-  var unique = (new Date()).valueOf();
-  title = title || unique.toString();
+      window.location.hash = url;
+    });
 
-  return title.toLowerCase();
-}
+    this.model.commit({
+      path: ['pages', pageTitle].join('/'),
+      content: this.doc.toMarkdown(),
+      message: 'Created ' + ['pages', pageTitle].join('/')
+    });
+  },
+  getSettingsFromEditor: function () {
+    var self = this,
+        remaining = this.editors.settings.doc.getValue(),
+        whitelist = this.settings.whitelist.map(function (v) {
+          var sel = '[name=' + v.name + ']', value;
+          if (v.type === 'boolean') {
+            sel += ':checked';
+          }
+
+          value = self.$(sel).val();
+          return [v.name, value].join(': ');
+        }).join('\n');
+
+    return [whitelist, remaining].join('\n');
+  },
+  getContentFromEditor: function () {
+    var content;
+
+    if (this.editors.content && this.editors.content.content) {
+      // ProseMirror is loaded as content editor
+      content = this.editors.content.getContent('markdown');
+    }
+    else if (this.editors.content) {
+      // CodeMirror is loaded as content editor
+      content = this.editors.content.doc.getValue();
+    }
+
+    return content;
+  },
+  getSettingsDisplayStyle: function (doc) {
+    if (doc.fileExt == 'md' || doc.fileExt == 'markdown') {
+      if (doc.frontMatter == '') return 'regular';
+      else return 'whitelist';
+    }
+    return 'only';
+  },
+  toIsoDateString: function (date) {
+    var d = (date) ? new Date(date) : new Date();
+    return d.toISOString().substring(0, 10)
+  },
+  filePathFromModel: function (model) {
+    return path = [
```

Comment:
is the `path =` assignment necessary here, or do you only want to return the joined string?